lib/fluent/config/literal_parser.rb

```diff
@@ -162,6 +163,8 @@ def eval_embedded_code(code)
         if @eval_context.nil?
           parse_error! "embedded code is not allowed in this file"
         end
+        hostname = Socket.gethostname
+        worker_id = ENV['SERVERENGINE_WORKER_ID'] || ''
```

Comment:
This approach shows warnings with verbose mode but I'm not sure how to avoid it...

Reply:
I can't see any warnings on my environment. Could you share the detail?

Reply:
See https://travis-ci.org/fluent/fluentd/jobs/327180965#L541

Reply:
Ah, I see. How about using `Binding#local_variable_set`?

Reply:
With `instance_eval`, `local_variable_set` doesn't work. Try this:

```rb
context = Kernel.binding
context.local_variable_set(:worker_id, 1)
p context.instance_eval('"foo#{worker_id}"')
```

Reply:
Can't we use `#eval` instead of `#instance_eval` here?

Reply:
I'm not sure it doesn't break existing behaviour or not.

@nurse Do you see any problem for using `eval` here?

Reply:
Just add to `code` seems simple.
```ruby
code = <<eom + code
hostname = Socket.gethostname
worker_id = ENV['SERVERENGINE_WORKER_ID'] || ''
eom
         @eval_context.instance_eval(code)
```